src/init.cpp

```diff
@@ -924,16 +924,6 @@ bool AppInitParameterInteraction(const ArgsManager& args, bool use_syscall_sandb
     fCheckBlockIndex = args.GetBoolArg("-checkblockindex", chainparams.DefaultConsistencyChecks());
     fCheckpointsEnabled = args.GetBoolArg("-checkpoints", DEFAULT_CHECKPOINTS_ENABLED);
 
-    if (args.IsArgSet("-minimumchainwork")) {
-        const std::string minChainWorkStr = args.GetArg("-minimumchainwork", "");
-        if (!IsHexNumber(minChainWorkStr)) {
-            return InitError(strprintf(Untranslated("Invalid non-hex (%s) minimum chain work value specified"), minChainWorkStr));
```

Comment:
In commit "Move ::nMinimumChainWork into ChainstateManager" (fa844d25eb1cfb081b81a322ef9ba656ffba3de9)

One not great thing about this change is it delays validation of this option until after `bitcoind` daemonizes. So `bitcoind` service could appear to start successfully returning exit code 0, and then immediately fail. I think more ideally `AppInitParameterInteraction` would continue to validate options and report errors early.

It may be a good idea to add a `util::Result<ChainstateManager::Options> MakeChainManOpts(ArgsManager&)` helper function that could be called from both `AppInitParameterInteraction` and `AppInitMain` to validate options earlier and use them later.

Reply:
This is also true for all other settings, for example `-maxmempool`, `-maxuploadtarget`, ... . `AppInitParameterInteraction` was only meant for setting *interaction*, not really for parsing. If there should be an effort to parse all settings before daemonization, I think this should be done separate from the changes here?

Reply:
> MakeChainManOpts

An alternative would be to do the parsing only once and put the result into some kind of context (the existing kernel context or a new init context)?

Reply:
Thanks, calling twice